community/cypher/cypher-compiler-2.2/src/main/scala/org/neo4j/cypher/internal/compiler/v2_2/SemanticChecker.scala

```diff
@@ -22,14 +22,20 @@ package org.neo4j.cypher.internal.compiler.v2_2
 import org.neo4j.cypher.internal.compiler.v2_2.ast.Statement
 
 class SemanticChecker(semanticCheckMonitor: SemanticCheckMonitor) {
-  def check(queryText: String, statement: Statement): SemanticState = {
+  def check(queryText: String, statement: Statement, offset: Option[InputPosition]): SemanticState = {
     semanticCheckMonitor.startSemanticCheck(queryText)
-    val SemanticCheckResult(semanticState, semanticErrors) = statement.semanticCheck(SemanticState.clean)
+    val SemanticCheckResult(semanticState, rawSemanticErrors) = statement.semanticCheck(SemanticState.clean)
 
     val scopeTreeIssues = ScopeTreeVerifier.verify(semanticState.scopeTree)
     if (scopeTreeIssues.nonEmpty)
       throw new InternalException(scopeTreeIssues.mkString(s"\n"))
 
+    val semanticErrors = rawSemanticErrors.map { error =>
```

Comment:
It's not really necessary to remap all the errors, since 12 lines further down only the first is thrown